python/src/aiconfig/model_parser.py

```diff
@@ -66,7 +66,6 @@ async def run(
         aiconfig: AIConfig,
         options: Optional["InferenceOptions"] = None,
         parameters: Dict = {},
-        **kwargs,  # TODO: Remove this, just a hack for now to ensure that it doesn't break
```

Comment:
We need to add `run_with_dependencies` arg here otherwise it will fail. Ex: https://github.com/lastmile-ai/aiconfig/blob/b7b9910cc7abbd9b08dbdfe634e25f98396afcef/python/src/aiconfig/editor/server/server.py#L381

It's weird that your test plan showed this working, this should've failed. Can you do `pip list | grep aiconfig` and make sure that your aiconfig is linked to local and not the production package? If it's not seeing `python-aiconfig` to a local path for you, you need to go to `aiconfig/python` and run `pip install -e .`

Reply:
@rossdanlm Yes! I am using the local package. I resintall the source package at 0.30 in the test demo video. 

As for why if doesn't fail is because `HuggingFaceTextSummarizationRemoteInference` inherits from `ParameterizedModelParser` and run with dependencies is taken care of for Parameterized Model Parser.

As for `ModelParser`, we don't have `run_with_dependencies` method implemented.

Reply:
https://github.com/lastmile-ai/aiconfig/blob/b7b9910cc7abbd9b08dbdfe634e25f98396afcef/python/src/aiconfig/editor/server/server.py#L381

I'm unsure about the mechanism behind aiconfig.run in determining which version of run to employ. Based on my understanding, it might be contingent upon the behavior defined in the model registry. Specifically, for HuggingFace models, my speculation is that it defaults to utilizing the ParameterizedModelParser.

Reply:
> As for why if doesn't fail is because HuggingFaceTextSummarizationRemoteInference inherits from ParameterizedModelParser and run with dependencies is taken care of for Parameterized Model Parser.

Sorry can you please test on models like `HuggingFaceImage2TextRemoteInference` (ex: image_caption prompt) which does not depend on `ParameterizedModelParser` and inherits directly from `ModelParser`. This is what I meant that I suspect this these should fail 

> I'm unsure about the mechanism behind aiconfig.run in determining which version of run to employ

All the model parsers that are text-based inputs, will inherit from `ParameterizedModelParser` which inherits from `ModelParser`. All the models that are attachment-based inputs (ex: `HuggingFaceImage2TextRemoteInference`) will inherit from `ModelParser` directly instead

Reply:
Updated the `ModelParser` as requested.